okio-files/src/commonMain/kotlin/okio/Path.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import okio.ByteString.Companion.EMPTY
+import okio.ByteString.Companion.encodeUtf8
+
+/**
+ * A hierarchical address on a file system. A path is an identifier only; a [Filesystem] is required
+ * to access the file that a path refers to, if any.
+ *
+ * Paths may be absolute or relative:
+ *
+ * * **Absolute paths** are prefixed with `/` and identify a location independent of any working
+ *   directory. An absolute path is either `/` or one or more non-empty segments, each of which is
+ *   prefixed with `/`. The strings `.` and `..` are never used as segments in absolute paths.
+ *
+ * * **Relative paths** are not prefixed with `/`. These are resolved against [Filesystem.cwd] when
+ *   accessed. The empty path `""` refers to [Filesystem.cwd]. A relative path is either the empty
+ *   path `""` or one or more non-empty `/`-separated segments. The string `.` is never used as a
+ *   segment in a relative path. A relative path may have `..` segments, but if they do they
+ *   precede all other segments.
+ *
+ * Sample Paths
+ * ------------
+ *
+ * | Path                   | Type       | Parent              |
+ * | :--------------------- | :--------- | :------------------ |
+ * | `/Users/jessewilson`   | Absolute   | `/Users`            |
+ * | `/Users`               | Absolute   | `/`                 |
+ * | `/`                    | Absolute   | null                |
+ * | `src/main/java`        | Relative   | `src/main`          |
+ * | `src/main`             | Relative   | `src`               |
+ * | `src`                  | Relative   | (empty)             |
+ * | (empty)                | Relative   | null                |
+ * | `../../src/main/java`  | Relative   | ` ../../src/main`   |
+ * | `../../src/main`       | Relative   | ` ../../src`        |
+ * | `../../src`            | Relative   | ` ../..`            |
+ * | `../..`                | Relative   | null                |
```

Comment:
Why is parent not relative to cwd? Because of the when accessed? This implies that a relative path is a persistent concept that can be reapplied to various cwd values with different results, correct?

Reply:
A relative path can be resolved against an absolute path, or perhaps another relative path!